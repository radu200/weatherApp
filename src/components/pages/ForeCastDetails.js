import React from "react";
import NavBar from "../navbar/NavBar";
import style from "./ForeCast.module.scss";
import ForeCast from "./ForeCast";
import PropTypes from "prop-types";

const ForeCastDetails = ({
  data,
  city
}) => {
  return (
    <>
      <NavBar />

      <div className={style.details_page_container}>
        {data && data.map(w => {
          return (
            <div key={w.id}>
              <ForeCast photo={w.icon} date={w.date} />
              <p data-testid='city' className={style.text_font}>{city}</p>
              <p data-testid='description'  className={style.text_font}>{w.description}</p>
              {w.temp_min && <p data-testid='minTemp'  className={style.text_font}> min temp : {w.temp_min}</p>}
              {w.temp_max && <p data-testid='maxTemp'  className={style.text_font}> max temp : {w.temp_max}</p>}
              {w.humidity && <p  data-testid='humidity'  className={style.text_font}> humidity : {w.humidity}</p>}
            </div>
          )
        })

        }
      </div>
    </>
  );
};

ForeCastDetails.propTypes = {
  data: PropTypes.array,
  city: PropTypes.string
}
export default ForeCastDetails;
