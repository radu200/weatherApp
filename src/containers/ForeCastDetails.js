import React from "react";
import PropTypes from 'prop-types'
import ForeCastDeatils from "../components/pages/ForeCastDetails";
import { connect } from "react-redux";
import { fetchWeather } from "../redux/weatherApp/operators";
import { getDaySelector, getCity } from "./../redux/weatherApp/selectors";

const ForeCastDetails  = ({data, city })  => {
  return (
      <>
        <ForeCastDeatils
          city={city}
          data={data}
        />
      </>
    );

  }

const mapState = state => {
  return {
    city: getCity(state),
    data:getDaySelector(state)
  };
};

ForeCastDetails.propTypes = {
  city:PropTypes.string,
  data:PropTypes.array
}

export default connect(mapState, { fetchWeather })(ForeCastDetails);
