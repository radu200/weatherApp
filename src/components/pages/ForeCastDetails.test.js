import React from "react";
import ForeCastDetails from "./ForeCastDetails";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("Details component", () => {
  const data = [
    {
      id: 1,
      icon: "10dn",
      date: "Friday, Nov 20",
      descriptions: "some description",
      temp_min: "10",
      temp_max: "12",
      humidity: "15"
    }
  ];

  let wrap;

  beforeEach(() => {
    wrap = shallow(<ForeCastDetails city="London" data={data} />);
  });

  it("should render ForeCastDetails with given props", () => {
    expect(shallowToJson(wrap)).toMatchSnapshot();
  });
   
  it("should render with city name", () => {
    expect(wrap.find(`[data-testid='city']`).text()).toMatch("London");
  });

  it("should render with city name", () => {
    expect(wrap.find(`[data-testid='maxTemp']`).text()).toMatch(
      " max temp : 12"
    );
  });

  it("should render with city name", () => {
    expect(wrap.find(`[data-testid='minTemp']`).text()).toMatch(
      " min temp : 10"
    );
  });


});
