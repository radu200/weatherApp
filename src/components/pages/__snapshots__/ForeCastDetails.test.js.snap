// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component should render ForeCastDetails with given props 1`] = `
<Fragment>
  <NavBar />
  <div
    className="details_page_container"
  >
    <div
      key="1"
    >
      <ForeCast
        date="Friday, Nov 20"
        photo="10dn"
      />
      <p
        className="text_font"
        data-testid="city"
      >
        London
      </p>
      <p
        className="text_font"
        data-testid="description"
      />
      <p
        className="text_font"
        data-testid="minTemp"
      >
         min temp : 
        10
      </p>
      <p
        className="text_font"
        data-testid="maxTemp"
      >
         max temp : 
        12
      </p>
      <p
        className="text_font"
        data-testid="humidity"
      >
         humidity : 
        15
      </p>
    </div>
  </div>
</Fragment>
`;
